### PR TITLE
Forward port release v0.5.2

### PR DESCRIPTION
**Pull request by joshuachp**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - Solve a panic when deserializing an empty BSON array.
 
+## [0.5.2] - 2024-01-30
```
Comment (harlem88) [inline CHANGELOG.md:42]: The release should be chronologically ordered, so even if it has minor version the should be insert up to `0.6.0` version.

File: astarte-device-sdk-derive/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [0.6.0] - 2023-07-05
 
+## [0.5.2] - 2024-01-30
```
Comment (harlem88) [inline astarte-device-sdk-derive/CHANGELOG.md:13]: same above

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/interfaces.rs [hunk 3]
```diff
@@ -271,6 +264,32 @@ impl FromIterator<(String, Interface)> for Interfaces {
...
+            write!(f, "{}:{}:{}", name, major, minor)?;
+        } else {
+            return Ok(());
+        }
+
+        for (name, interface) in iter {
+            let major = interface.version_major();
+            let minor = interface.version_minor();
+            write!(f, ";{}:{}:{}", name, major, minor)?;
+        }
```
Comment (rgallor) [inline src/interfaces.rs:287]: What do yopu think of the following?
```suggestion
        let Some((name, interface)) = iter.next() else {
            return Ok(());
        }

        let mut major = interface.version_major();
        let mut minor = interface.version_minor();
        write!(f, "{}:{}:{}", name, major, minor)?;

        for (name, interface) in iter {
            major = interface.version_major();
            minor = interface.version_minor();
            write!(f, ";{}:{}:{}", name, major, minor)?;
        }
```

**Review by rgallor — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (joshuachp) [reply CHANGELOG.md]: Done

File: astarte-device-sdk-derive/CHANGELOG.md (hunk 2 above)
Comment (joshuachp) [reply astarte-device-sdk-derive/CHANGELOG.md]: Done

File: src/interfaces.rs (hunk 3 above)
Comment (joshuachp) [reply src/interfaces.rs:287]: This is the forward port from 1.59 this syntax was not supported in that rust version

Comment (joshuachp) [reply src/interfaces.rs:287]: I will add a commit to refactor this

Comment (joshuachp) [reply src/interfaces.rs]: Done

**Review by rgallor — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**